src/cleanfigure.m

```diff
@@ -347,104 +349,161 @@ function simplifyLine(meta, handle, targetResolution)
     yData = get(handle, 'YData');
     zData = get(handle, 'ZData');
 
-    if ~isempty(zData)
-        % TODO: 3d simplificattion of frontal 2d projection
-        return;
-    end
-    if isempty(xData) || isempty(yData)
+    if numel(xData) <= 2 || numel(xData) <= 2
         return;
     end
-    if numel(xData) <= 2
-        return;
+
+    % Check whether this is a 3D plot
+    if numel(zData) >1
+        is3D = true;
+    else
+        is3D = false;
     end
```

Comment:
We should not rely on data to test 3D of the axis. Sometimes, there is ZData but it's all ones, yet the view is 2D.

I thought we had `isAxis3D` in the `test/private` but it is only in `src\private`. Please, copy that function into `test/private` and use that.